openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -217,7 +219,10 @@ def create_account_with_params(request, params):
     if skip_email:
         registration.activate()
     else:
-        compose_and_send_activation_email(user, profile, registration)
+        if is_testing_environment():
+            compose_and_send_activation_email(user, profile, registration)
+        else:
+            compose_and_send_adg_activation_email(user, registration.activation_key)
```

Comment:
There are in total 4 occurrences of `compose_and_send_activation_email` in the project, which means these emails are sent from three more places.

My suggestion would be to inject out code inside `compose_and_send_activation_email` rather than from where this method is called. It will cater to all possibilities.

Reply:
I choose to inject our function here because in edX 'compose_and_send_activation_email' there are nested function and logic that we don't need so as per my opinion it is better here.